sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlCliTest.java

```diff
@@ -41,13 +50,178 @@ public void testExecute_createTextTable() throws Exception {
+
+    BeamSqlCli cli = new BeamSqlCli()
+        .metaStore(metaStore);
+    cli.execute(
+        "create table person (\n"
+        + "id int COMMENT 'id', \n"
+        + "name varchar COMMENT 'name', \n"
+        + "age int COMMENT 'age', \n"
+        + "tags VARCHAR ARRAY \n"
+        + ") \n"
+        + "TYPE 'text' \n"
+        + "COMMENT '' LOCATION '/home/admin/orders'"
+    );
+    Table table = metaStore.getTable("person");
+    assertNotNull(table);
+    assertEquals(
+        Stream
+            .of(
+                Field.of("id", INTEGER).withDescription("id").withNullable(true),
+                Field.of("name", VARCHAR).withDescription("name").withNullable(true),
+                Field.of("age", INTEGER).withDescription("age").withNullable(true),
+                Field.of("tags",
+                         ARRAY.type().withCollectionElementType(VARCHAR)).withNullable(true))
+            .collect(toSchema()),
+        table.getSchema());
+  }
+
+  @Test
+  public void testExecute_createTableWithPrefixArrayField() throws Exception {
+    InMemoryMetaStore metaStore = new InMemoryMetaStore();
+    metaStore.registerProvider(new TextTableProvider());
+
+    BeamSqlCli cli = new BeamSqlCli()
+        .metaStore(metaStore);
+    cli.execute(
+        "create table person (\n"
+        + "id int COMMENT 'id', \n"
+        + "name varchar COMMENT 'name', \n"
+        + "age int COMMENT 'age', \n"
+        + "tags ARRAY<VARCHAR>, \n"
+        + "matrix ARRAY<ARRAY<INTEGER>> \n"
+        + ") \n"
+        + "TYPE 'text' \n"
+        + "COMMENT '' LOCATION '/home/admin/orders'"
+    );
+    Table table = metaStore.getTable("person");
+    assertNotNull(table);
+    assertEquals(
+        Stream
+            .of(
+                Field.of("id", INTEGER).withDescription("id").withNullable(true),
+                Field.of("name", VARCHAR).withDescription("name").withNullable(true),
+                Field.of("age", INTEGER).withDescription("age").withNullable(true),
+                Field.of("tags",
+                         ARRAY.type().withCollectionElementType(VARCHAR)).withNullable(true),
+                Field.of("matrix",
+                         ARRAY.type().withCollectionElementType(
+                             ARRAY.type().withCollectionElementType(INTEGER))).withNullable(true))
+            .collect(toSchema()),
+        table.getSchema());
+  }
+
+  @Test
+  public void testExecute_createTableWithPrefixMapField() throws Exception {
+    InMemoryMetaStore metaStore = new InMemoryMetaStore();
+    metaStore.registerProvider(new TextTableProvider());
+
+    BeamSqlCli cli = new BeamSqlCli()
+        .metaStore(metaStore);
+    cli.execute(
+        "create table person (\n"
+        + "id int COMMENT 'id', \n"
+        + "name varchar COMMENT 'name', \n"
+        + "age int COMMENT 'age', \n"
+        + "tags MAP<VARCHAR, VARCHAR>, \n"
+        + "nestedMap MAP<INTEGER, MAP<VARCHAR, INTEGER>> \n"
```

Comment:
<!--new_thread; commit:acf50611c0f2b329b63e3a21ccc9e54db51e5d97; resolved:0-->
Can you also test:

 - maps inside arrays
 - arrays inside maps
 - rows inside arrays
 - rows inside maps
 - maps inside rows
 - arrays inside rows
 - nested nested rows

... etc ...

At some point we might want to use QuickCheck/SmallCheck; that is one of the best ways to test valid syntax.

Reply:
Will do